llvm/test/Analysis/CostModel/AArch64/shuffle-extract.ll

```diff
@@ -96,8 +102,129 @@ define void @extract_half() {
   ret void
 }
 
-define void @extract_qtr() {
-; CHECK-LABEL: 'extract_qtr'
+; Similar as above, but for a 256-bit SVE core.
+; Vectors smaller than 256-bit aren't tested again.
+define void @extract_half_vscale2() vscale_range(2,2) {
```

Comment:
These tests define a vscale range but I don't see any usage of vscale in the changes or instructions in the test. Could you explain how the vscale range interacts with these changes?